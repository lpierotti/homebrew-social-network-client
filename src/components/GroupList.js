import React from 'react'
import { connect } from 'react-redux'
import { getGroups } from '../actions/users'
import FollowDisplay from './FollowDisplay'
import { Link } from 'react-router-dom'
import { Segment, Card, Divider } from 'semantic-ui-react'

class GroupList extends React.Component {

	componentDidMount() {
		this.props.getGroups()
	}

	render() {
		return (
			<div style={{maxWidth: '1250px', margin: 'auto'}}>
			<Card.Group itemsPerRow={2}>
				{this.props.userGroups.map((group, index) => {
					return (
						<div key={index} style={{maxWidth: '600px', margin: '10px'}}>
							<Segment inverted style={{backgroundColor: 'rgba(95, 122, 83, 0.81)'}} raised={true}>
								<Link to={`/group/${group.id}`}><h2 style={{color: 'black'}}>{group.name}</h2></Link>
								<Divider />
								<h4>{group.description}</h4>
								<Segment >
									<Card.Group itemsPerRow={3} textAlign='center'>
										{group.members.map((member, index) => <FollowDisplay  key={index} data={member} />)}
									</Card.Group>
								</Segment>
							</Segment>
						</div>
					)
				})}
			</Card.Group>
			</div>
		)
	}
}	

function mapStateToProps(state) {
	return {
		userGroups: state.users.userGroups
	}
}
function mapDispatchToProps(dispatch) {
	return {
		getGroups: () => {dispatch(getGroups())}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(GroupList)