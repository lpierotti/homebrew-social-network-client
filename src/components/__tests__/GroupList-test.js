import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { GroupList } from '../GroupList';
import renderer from 'react-test-renderer';

describe('GroupList component', () => {

	it('renders correctly', () => {
		const tree = renderer.create(
			<GroupList userGroups={[]} getGroups={() => {}}/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	})
})