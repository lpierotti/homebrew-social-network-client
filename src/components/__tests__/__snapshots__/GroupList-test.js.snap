// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupList component renders correctly 1`] = `
<div
  style={
    Object {
      "margin": "auto",
      "maxWidth": "1250px",
    }
  }
>
  <div
    className="ui two cards"
  />
</div>
`;
